src/simcore_service_deployment_agent/portainer.py

```diff
@@ -189,5 +189,29 @@ async def update_stack(
         )
         log.debug("updated stack: %s", data)
     except asyncio.exceptions.TimeoutError as err:
-        print("ERROR")
-        print(f"{err}")
+        log.error(str(err))
+
+
+async def delete_stack(
+    base_url: URL,
+    app_session: ClientSession,
+    bearer_code: str,
+    stack_id: str,
+    endpoint_id: int,
+):  # pylint: disable=too-many-arguments
+    log.debug("deleting stack %s", base_url)
+    if endpoint_id < 0:
```

Comment:
Q: I remember introducing that a very long time ago. do you actually still call it with an endpoint that is not negative?

Reply:
I guess `endpoint_id < 0` means the endpoint is not determined yet, this seems to be propagated throughtout the code